File: pet-sitting-front/libs/services/src/lib/services/models/user.model.ts
import { AccomodationTypeEnum } from "./accomodation.enum";
import { RoleEnum } from "./role.enum";

export interface UserModel {
    id?: number;
    lastName?: string;
    firstName?: string;

    about?: string;
    dailyPrice?: number;
    image? : string;
    accomodationType?: AccomodationTypeEnum;
    address?: string;
    postalCode?: string;
    city?: string;
    pays?:string;
    livingSpace?:number;
    hasGaden?: boolean;
    hasVehicule ?: boolean;
    telephone?: string;
    email?: string;
    password?:string
    role ?: RoleEnum,
    dateOfBirth? : Date;
    status ?: 'AVAILABLE' | 'NOTAVAILABLE' | 'BANNED';

  }

File: pet-sitting-backend/src/main/java/fr/aston/petSitting/modele/UserModel.java
/**
 * 
 */
package fr.aston.petSitting.modele;

import java.math.BigDecimal;
import java.util.Date;

import fr.aston.petSitting.entities.AccomodationTypeEnum;
import fr.aston.petSitting.entities.RoleEnum;
import fr.aston.petSitting.entities.StatusEnum;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author F_CHAMAR
 *
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserModel {

	private int id;
	private AccomodationTypeEnum accomodationType;
	private String address;
	private String city;
	private Date dateOfBirth;
	private String email;
	private String firstName;
	private Boolean hasGarden;
	private Boolean hasVehicule;
	private String lastName;
	private BigDecimal livingSpace;
	private String password;
	private String pays;
	private String postalCode;
	private String about;
	private RoleEnum role;
	@Enumerated(EnumType.STRING)
	private StatusEnum status;
	private String telephone;
}
